lib/Button/index.js

```diff
@@ -33,42 +36,31 @@ class Button extends Component {
     };
 
     this.handleOnClick = this.handleOnClick.bind(this);
+    this.getTypeClasses = this.getTypeClasses.bind(this);
   }
 
   handleOnClick() {
     this.setState({ disabled: this.props.disableOnClick });
     this.props.clickHandler();
   }
 
+  getTypeClasses() {
```

Comment:
I see what you mean, there's a plus and a down side to this approach. On one hand it's a much cleaner (and more maintainable) approach; on the other we lost the ability to quickly scan the possibilities that are supported by our styleguide.

Maybe keeping the documentation comments at the top with all the possible options is a way to mitigate this, or to declare the component's usage for all cases in the actual style guide could also be a way.